Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -74,16 +78,471 @@ theorem algebraMapCLM_toLinearMap : (algebraMapCLM R A).toLinearMap = Algebra.li
   rfl
 #align algebra_map_clm_to_linear_map algebraMapCLM_toLinearMap
 
+/-- If `R` is a discrete topological ring, then any topological ring `S` which is an `R`-algebra
+  is also a topological `R`-algebra. -/
+instance DiscreteTopology.continuousSMul [TopologicalSemiring A] [DiscreteTopology R] :
+    ContinuousSMul R A := continuousSMul_of_algebraMap _ _ continuous_of_discreteTopology
+
 end TopologicalAlgebra
 
-section TopologicalAlgebra
+section TopologicalAlgebra --
+section
 
 variable {R : Type*} [CommSemiring R]
 variable {A : Type u} [TopologicalSpace A]
 variable [Semiring A] [Algebra R A]
 
 #align subalgebra.has_continuous_smul SMulMemClass.continuousSMul
 
+end
+section
+
+variable (R : Type*) [CommSemiring R] [TopologicalSpace R] [TopologicalSemiring R]
+  (A : Type*) [Semiring A] [TopologicalSpace A] [TopologicalSemiring A]
+
+/-- Continuous algebra homomorphisms between algebras. We only put the type classes that are
+  necessary for the definition, although in applications `M` and `B` will be topological algebras
+  over the topological ring `R`. -/
+structure ContinuousAlgHom (R : Type*) [CommSemiring R] (A : Type*) [Semiring A]
+    [TopologicalSpace A] (B : Type*) [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+    extends A →ₐ[R] B where
+  cont : Continuous toFun := by continuity
+
+/-- `ContinuousAlgHomClass F R A B` asserts `F` is a type of bundled continuous `R`-agebra
+  homomorphisms `A → B`. -/
+class ContinuousAlgHomClass (F : Type*) (R : outParam (Type*)) [CommSemiring R]
+    (A : outParam (Type*)) [Semiring A] [TopologicalSpace A]
+    (B : outParam (Type*)) [Semiring B] [TopologicalSpace B][Algebra R A]
+    [Algebra R B] [FunLike F A B]
+    extends AlgHomClass F R A B, ContinuousMapClass F A B : Prop
+attribute [inherit_doc ContinuousAlgHom] ContinuousAlgHom.cont
+
+@[inherit_doc]
+notation:25 A " →A[" R "] " B => ContinuousAlgHom R A B
+
+namespace ContinuousAlgHom
+
+section Semiring
+
+variable {R} {A}
+variable [TopologicalSpace R] [TopologicalSemiring R]
+
+variable {B : Type*} [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+
+attribute [coe] ContinuousAlgHom.toAlgHom
+/-- Coerce continuous algebra morphisms to algebra morphisms. -/
+instance AlgHom.coe : Coe (A →A[R] B) (A →ₐ[R] B) := ⟨toAlgHom⟩
```

Comment:
This creates two non-defeq coercions from `A →A[R] B` to `A →ₐ[R] B`. There's this one, that you're adding, but with the instance `continuousAlgHomClass` below (which we should definitely have), we get the one passing through `AlgHomClass`, namely [AlgHomClass.toAlgHom](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Hom.html#AlgHomClass.toAlgHom) via [AlgHomClass.coeTC](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Hom.html#AlgHomClass.coeTC). We have a preference for the latter, so please delete this one; this may require some re-jiggering of your code below.